fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -622,13 +622,15 @@ private void handleQueryStmt() throws Exception {
         RowBatch batch;
         MysqlChannel channel = context.getMysqlChannel();
         boolean isOutfileQuery = queryStmt.hasOutFileClause();
-        if (!isOutfileQuery) {
-            sendFields(queryStmt.getColLabels(), queryStmt.getResultExprs());
-        }
+        boolean isSendFields = false;
```

Comment:
So the reason is that the `fields` has been sent but no data is sent following?

Reply:
Yes, fields has been sent, but query may fail because of timeout. In this case, client will get a fields packet, and error packet, but some language mysql driver cannot recognize the error, and returns a success and no rows result.